lnbits/extensions/boltcards/lnurl.py

```diff
@@ -0,0 +1,204 @@
+import base64
+import hashlib
+import hmac
+import json
+import secrets
+from http import HTTPStatus
+from io import BytesIO
+from typing import Optional
+
+from embit import bech32, compact
+from fastapi import Request
+from fastapi.param_functions import Query
+from fastapi.params import Depends, Query
+from lnurl import Lnurl, LnurlWithdrawResponse
+from lnurl import encode as lnurl_encode  # type: ignore
+from lnurl.types import LnurlPayMetadata  # type: ignore
+from loguru import logger
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import HTMLResponse
+
+from lnbits.core.services import create_invoice
+from lnbits.core.views.api import pay_invoice
+
+from . import boltcards_ext
+from .crud import (
+    create_hit,
+    get_card,
+    get_card_by_otp,
+    get_card_by_uid,
+    get_hit,
+    get_hits_today,
+    spend_hit,
+    update_card,
+    update_card_counter,
+    update_card_otp,
+)
+from .models import CreateCardData
+from .nxp424 import decryptSUN, getSunMAC
+
+###############LNURLWITHDRAW#################
+
+# /boltcards/api/v1/scan?p=00000000000000000000000000000000&c=0000000000000000
+@boltcards_ext.get("/api/v1/scan/{card_uid}")
```

Comment:
The `p` query parameter is the encrypted `card_uid` together with a counter. Is it a good idea to expose the unencrypted `{card_uid}` in the path parameter (in the static part of lnurlw)? This could be a security issue. I would recommend to use another (database internal) id as path parameter instead of the card_uid!

Reply:
Added.